### PR TITLE
Synonyms of species during standardisation

### LINKED ISSUE #969: `standardise_footprint` doesn't convert species to synonyms
### What happened?

Adding footprints with a species like `hfo-1234zee` works, but if you search by species `hfo-1234zee` returns no results. This is because `_base_search` converts `hfo-1234zee` to its synonym `hfo1234zee`. However, when standardising the data, the species is stored as `hfo-1234zee`.

### What did you expect to happen?

Searching for data by species should work.

### Minimal Complete Verifiable Example

_No response_

### Relevant log output

_No response_

### Anything else we need to know?

_No response_

### Environment

<details>

</details>

### PR DESCRIPTION
**Pull request by SutarPrasad**
* **Summary of changes** (Bug fix, feature, docs update, ...)
  Added call to synonyms in read_file() function of all data_type classes.

* **Please check if the PR fulfills these requirements**

- [x] Closes #969 #1003 (Replace xxxx with the Github issue number), 
- [x] [Tests added and passed](https://docs.openghg.org/development/python_devel.html#testing) if fixing a bug or adding a new feature
- [x] All code checks passing - `black --line-length 110` run over code, `mypy` and `flake8` not showing any errors
- [x] Added an entry in the latest `CHANGELOG.md` file if fixing a bug or adding a new feature
- [x] Added any new requirements to `requirements.txt` and `recipes/meta.yaml`

Not required: - [ ] Documentation and tutorials updated/added

### CODE REVIEW

File: openghg/store/_footprints.py [hunk 1]
```diff
@@ -301,6 +301,7 @@ def read_file(
             species = "inert"
         else:
             species = clean_string(species)
+            species = synonyms(species)
```
Comment (rt17603) [inline openghg/store/_footprints.py:304]: Does this do the right thing if `species="inert"` i.e. leave it unchanged? I assume if the tests pass it does but worth checking that explicitly.

**Review by rt17603 — COMMENTED** [review]
Looks good and a really useful tidy up to include.

Would be useful to add a test for the know fail case mentioned in the original issue #969 specifically for the search interface for a gas like "hfo-1234zee" (i.e. with a "-" in it).

**Comment by rt17603** [comment]
Hi Prasad, thanks for adding the test to check that "inert" is standardised correctly but the test I was actually mentioned:

> Would be useful to add a test for the known fail case mentioned in the original issue https://github.com/openghg/openghg/issues/969 specifically for the search interface for a gas like "hfo-1234zee" (i.e. species name with a "-" in it).

Basically adding a check for a footprint file which has "hfo-1234zee" as a species to make sure this can be standardised and retrieved as this was raised in the original issue.

**Comment by SutarPrasad** [comment]
> Hi Prasad, thanks for adding the test to check that "inert" is standardised correctly but the test I actually mentioned was:
> 
> > Would be useful to add a test for the known fail case mentioned in the original issue #969 specifically for the search interface for a gas like "hfo-1234zee" (i.e. species name with a "-" in it).
> 
> Basically adding a check for a footprint file which has "hfo-1234zee" as a species to make sure this can be standardised and retrieved as this was raised in the original issue.



> Hi Prasad, thanks for adding the test to check that "inert" is standardised correctly but the test I actually mentioned was:
> 
> > Would be useful to add a test for the known fail case mentioned in the original issue #969 specifically for the search interface for a gas like "hfo-1234zee" (i.e. species name with a "-" in it).
> 
> Basically adding a check for a footprint file which has "hfo-1234zee" as a species to make sure this can be standardised and retrieved as this was raised in the original issue.

Oh apologies from myside. Will quickly compress a file and add test for it. This will also take care of the bug.

**Comment by SutarPrasad** [comment]
> Hi Prasad, thanks for adding the test to check that "inert" is standardised correctly but the test I actually mentioned was:
> 
> > Would be useful to add a test for the known fail case mentioned in the original issue #969 specifically for the search interface for a gas like "hfo-1234zee" (i.e. species name with a "-" in it).
> 
> Basically adding a check for a footprint file which has "hfo-1234zee" as a species to make sure this can be standardised and retrieved as this was raised in the original issue.

Added the test which standardises the file and also retrieves with expected synonym name. Also checks for data_type as the search operation is only performed using species name(extra validation).

File: tests/store/test_footprints.py [hunk 2]
```diff
@@ -697,3 +697,39 @@ def test_optional_metadata():
...
+    )
+
+    # Get the footprints data
+    footprint_results = search(species=species)
+
+    footprint_obs = footprint_results.retrieve_all()
+    footprint_metadata = footprint_obs.metadata
+
+    assert footprint_obs.data is not None
+    assert "hfo1234ze" in footprint_metadata["species"]
```
Comment (rt17603) [inline tests/store/test_footprints.py:734]:
```suggestion
    assert footprint_metadata["species"] == "hfo1234zee"
```

**Review by rt17603 — APPROVED** [review]
Looks all good to me - thanks for getting this in, this will is a good loose end to tie up.

**Pull request merged**